twenty-first/src/util_types/mmr/mmr_trait.rs

```diff
@@ -39,42 +39,129 @@ impl LeafMutation {
     }
 }
 
+/// A “Merkle Mountain Range” (MMR) is a data structure for storing lists of
+/// hashes. Similar to [Merkle trees](crate::prelude::MerkleTree), they enable
+/// efficient set inclusion proofs. Unlike Merkle trees, the number of leafs of
+/// an MMR is not required to be a power of 2.
+///
+/// One valid perspective of this data structure is to see it as a list of
+/// Merkle trees. Each of the Merkle tree's roots is one peak of the MMR.
+/// In order to keep the number of peaks to a minimum, any two neighboring
+/// Merkle trees of the same height are merged, which results in a single tree
+/// that has one additional “tier”.
```

Comment:
I guess "neighboring" is superfluous here. Any two Merkle trees of the same height are merged.

Reply:
"tier", I guess, can also be described as height.